contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -0,0 +1,190 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-exchange/contracts/src/interfaces/IExchange.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibOrder.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibMath.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-asset-proxy/contracts/src/libs/LibAssetData.sol";
+
+
+contract OrderValidationUtils is
+    LibAssetData,
+    LibMath
+{
+    using LibBytes for bytes;
+
+    // solhint-disable var-name-mixedcase
+    IExchange internal EXCHANGE;
+    bytes internal ZRX_ASSET_DATA;
+    address internal ERC20_PROXY_ADDRESS;
+    // solhint-enable var-name-mixedcase
+
+    constructor (address _exchange, bytes memory _zrxAssetData)
+        public
+    {
+        EXCHANGE = IExchange(_exchange);
+        ZRX_ASSET_DATA = _zrxAssetData;
+        ERC20_PROXY_ADDRESS = EXCHANGE.getAssetProxy(ERC20_PROXY_ID);
+    }
+
+    /// @dev Fetches information for order and maker/taker of order.
+    /// @param order The order structure.
+    /// @param signature Proof that order has been created by maker.
+    /// @return OrderInfo, the remaining amount fillable by the taker, and validity of signature for given order.
```

Comment:
A general thought about Solidity function comments. These comments are going to become **the** canonical docs we will be showing them to **all** developers browsing our docs. 

<img width="982" alt="Screen Shot 2019-06-04 at 10 45 08" src="https://user-images.githubusercontent.com/2151492/58869300-db6bfa00-86b5-11e9-8d15-d20bb9fcedbb.png">

And so it's really important that they be fully-fleshed out and robust. Until now, they have been more minimal with the expectation that those reading them will also read the source-code, but this won't be the case for the majority of the readers. Therefore, I would appreciate it if you guys could spend more time making the function comments more robust and over-explain things. 

E.g.:

```
 /// @dev Fetches all order-relevant information that is needed to validate that the supplied 0x order is
///  fillable and for how much it can be filled.
/// @param order The order structure.
/// @param signature Order signature corresponding to the supplied order. If the signature does not need to be
/// validated, `0x01` can be supplied for the signature to always return `false`.
/// @return OrderInfo, the remaining amount fillable by the taker, and validity of signature for the given order. 
/// If the `takerAssetData` encodes for multiple assets, `fillableTakerAssetAmount` will represent a "scaled"
/// amount, meaning it must be multiplied by all the individual asset amounts within the multiAsset to get the final
/// amount of each asset that can still be filled.
```